sdk/core/azure-core/inc/azure/core/credentials/credentials.hpp

```diff
@@ -60,6 +60,12 @@ namespace Azure { namespace Core { namespace Credentials {
      *
      */
     std::string TenantId;
+
+    /**
+     * @brief Use credentials cache.
+     *
+     */
+    bool CacheCredentials = true;
```

Comment:
This is public API surface. I don't think other languages have this (although this is not necessarily a blocking factor).

The name should be something like `AllowCached`.

But I think you may not need this. We have similar tests in Azure Identity, and to bypass caching, we set `MinimumExpiration` to some extra large value:
https://github.com/Azure/azure-sdk-for-cpp/blob/dffa3edb2af2ab2673d0baad85cf4f49b7be9713/sdk/identity/azure-identity/test/ut/credential_test_helper.cpp#L142

I think, you can even use `DateTime::duration::max()`.

This effectively means, "if current token expires in less than $MinimumExpiration, do not use the cached value but get a new value instead, and cache it"